### PR TITLE
add unit test for DropRunAsGroupField,GetPodCondition and UpdatePodCondition

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for DropRunAsGroupField,GetPodCondition and UpdatePodCondition
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by php-coder** [comment]
/ok-to-test

@krmayankk JFYI

File: pkg/api/pod/util_test.go [hunk 1]
```diff
@@ -338,3 +339,282 @@ func TestDropAlphaVolumeDevices(t *testing.T) {
...
+			t.Errorf("case[%d]:%s: Expect condition: %#v, Got condition: %#v", i, test.name, *test.condition, *condition)
+		}
+	}
+}
+
+func isEqualCondition(leftcondition, rightcondition *api.PodCondition) bool {
+	if leftcondition == nil || rightcondition == nil {
+		if leftcondition == rightcondition {
+			return true
+		} else {
```
Comment (fisherxu) [inline pkg/api/pod/util_test.go]: no need the else.

Comment (hanxiaoshuai) [reply pkg/api/pod/util_test.go]: Done

File: pkg/api/pod/util_test.go [hunk 2]
```diff
@@ -338,3 +339,281 @@ func TestDropAlphaVolumeDevices(t *testing.T) {
...
+					SecurityContext: &api.SecurityContext{
+						RunAsGroup: &groupID,
+					},
+				},
+			},
+		},
+	}
+
+	// Enable feature RunAsGroup
+	err1 := utilfeature.DefaultFeatureGate.Set("RunAsGroup=true")
```
Comment (hzxuzhonghu) [inline pkg/api/pod/util_test.go]: nit: err looks better

File: pkg/api/pod/util_test.go [hunk 3]
```diff
@@ -338,3 +339,281 @@ func TestDropAlphaVolumeDevices(t *testing.T) {
...
+	if err1 != nil {
+		t.Fatalf("Failed to enable feature gate for RunAsGroup: %v", err1)
+	}
+
+	// now test dropping the fields - should not be dropped
+	DropDisabledAlphaFields(&testPod.Spec)
+
+	// check to make sure RunAsGroup is still present
```
Comment (hzxuzhonghu) [inline pkg/api/pod/util_test.go]: what if `RunAsGroup ` not exist? Have not seen any behavior.

Comment (hanxiaoshuai) [reply pkg/api/pod/util_test.go]: Done

File: pkg/api/pod/util_test.go (hunk 2 above)
Comment (hanxiaoshuai) [reply pkg/api/pod/util_test.go]: Done

**Comment by sttts** [comment]
@hanxiaoshuai  please don't assign for approval before it has a lgtm. @thockin and @liggitt will thank you :)

**Pull request closed without merging**